crates/bevy_render/src/render_phase/mod.rs

```diff
@@ -89,6 +89,29 @@ impl<I: PhaseItem> RenderPhase<I> {
             draw_function.draw(world, render_pass, view, item);
         }
     }
+
+    /// Renders all [`PhaseItem`]s in the provided `range` (based on their index in `self.items`) using their corresponding draw functions.
+    pub fn render_range<'w>(
+        &self,
+        render_pass: &mut TrackedRenderPass<'w>,
+        world: &'w World,
+        view: Entity,
+        range: Range<usize>,
+    ) {
+        let draw_functions = world.resource::<DrawFunctions<I>>();
+        let mut draw_functions = draw_functions.write();
+        draw_functions.prepare(world);
+
+        for item in self
+            .items
+            .iter()
+            .skip(range.start)
+            .take(range.end - range.start)
```

Comment:
`item` is `Vec` here, so this should work. The difference being this skips when range is partially outside of `items`. (say `items.len() == 3` and `range = 2..5`)
```suggestion
        for item in self.items.get(range).iter().flatten()
```

Reply:
Couldn't get that to work exactly because of the nested reference in the type, so I ended up adding a `.expect()` instead, which will panic if you use an out-of-bounds range, which might also be less confusing than just silently ignoring the range if it's out of bounds.